src/Installers/Windows/WindowsHostingBundle/Bundle.wxs

```diff
@@ -57,13 +57,126 @@
         <!-- Bundle variables -->
         <Variable Name="BundleNameShort" Value="$(var.BundleNameShort)"/>
         <Variable Name="BundleNameSub" Value="$(var.BundleNameSub)"/>
-        <Variable Name="OPT_NO_ANCM" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_FTS" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_LTS" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_SHAREDFX" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_RUNTIME" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_X86" Value="0" bal:Overridable="yes"/>
-        <Variable Name="OPT_NO_SHARED_CONFIG_CHECK" Value="0" bal:Overridable="yes" />
+        <Variable Name="OPT_NO_ANCM" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_FTS" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_LTS" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_SHAREDFX" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_RUNTIME" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_X86" bal:Overridable="yes"/>
+        <Variable Name="OPT_NO_SHARED_CONFIG_CHECK" bal:Overridable="yes" />
+
+        <!-- Use values from the registry for switches, if the registry values are set & the user hasn't passed any switches from the command line -->
```

Comment:
Explain how some registry values could be set while others are not. This seems significantly more complicated than it would be w/ a single check for either the registry key or one value.

Reply:
The only way that one registry value could be set and not another would be if the user manually modified the registry. But, having to create this conditional based on a registry search is sort of a WiX quirk - there isn't a direct way to assign a variable value based off of user input in a bundle. You can, however, conditionally set a variable to the value of a registry key based off of user input. We could just do this conditional check once based off of the existence of just one arbitrary registry key, but the cost of this is low and its purpose is more explicit. @joeloff do you agree?

Reply:
Yup, and for the initial deployment from MU, there would be no existing options recorded in the registry, so we want a default of 0, but if we set that initially and the user overrides it, then the registry search will win over the user's choice.